tests/Unit/Adapter/ReferenceFinder/Util/ContainerTypeResolverTest.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+namespace Phpactor\Indexer\Tests\Unit\Adapter\ReferenceFinder\Util;
+
+use Generator;
+use PHPUnit\Framework\TestCase;
+use Phpactor\Indexer\Adapter\ReferenceFinder\Util\ContainerTypeResolver;
+use Phpactor\Indexer\Tests\IntegrationTestCase;
+use Phpactor\TestUtils\ExtractOffset;
+
+class ContainerTypeResolverTest extends IntegrationTestCase
```

Comment:
This class has less responsiblity than it did originally as I moved the main logic to WorseReflection